merge nov 20th - include custom cmd advanced ci (#6346)
fishtown-analytics/docs.getdbt.com: website/docs/docs/deploy/ci-jobs.md

    - **Linting**<Lifecycle status="beta" /> &mdash; Enable this option for dbt to [lint the SQL files](/docs/deploy/continuous-integration#sql-linting) in your project as the first step in `dbt run`. If this check runs into an error, dbt can either **Fail job run** or **Continue running job**. 
    - **Run compare changes**<Lifecycle status="enterprise" /> &mdash; Enable this option to compare the last applied state of the production environment (if one exists) with the latest changes from the pull request, and identify what those differences are. To enable record-level comparison and primary key analysis, you must add a [primary key constraint](/reference/resource-properties/constraints) or [uniqueness test](/reference/resource-properties/data-tests#unique). Otherwise, you'll receive a "Primary key missing" error message in dbt Cloud.

      To review the comparison report, navigate to the [Compare tab](/docs/deploy/run-visibility#compare-tab) in the job run's details. A summary of the report is also available from the pull request in your Git provider (see the [CI report example](#example-ci-report)). 

      :::info Optimization tip 

>       When you enable the **Run compare changes** checkbox, you can customize the comparison command to optimize your CI job. For example, if you have large models that take a long time to compare, you can exclude them to speed up the process using the `--exclude` flag. Use [custom commands](/docs/deploy/job-commands#advanced-ci-compare-changes) to modify the selection criteria.

Author: ```suggestion
      When you enable the **Run compare changes** checkbox, you can customize the comparison command to optimize your CI job. For example, if you have large models that take a long time to compare, you can exclude them to speed up the process using the [`--exclude` flag](/reference/node-selection/exclude). Use [custom commands](/docs/deploy/job-commands#advanced-ci-compare-changes) to modify the selection criteria.
```

>       :::

Reviewer: I wonder if event_time belongs in this same callout box. That way, we could give tips for both model selection and event time, and both can link out to separate pages. 

@joellabes should be able to help with event time.

Author: i thought the user can't do anything with `event_time` yet though bc the `--sample` flag (name tbd) won't be available until feb 2025?

Reviewer 2: Today, event_time is used if set to do an "apples with apples" comparison. If you've hand-rolled a sampling thing in CI (e.g. [with this](https://docs.getdbt.com/best-practices/best-practice-workflows#limit-the-data-processed-when-in-development)) then setting event_time will let advanced CI filter to only the overlapping date ranges from the two tables

>       When you enable the **Run compare changes** checkbox, you can customize the comparison command to optimize your CI job. For example, if you have large models that take a long time to compare, you can exclude them to speed up the process using the [`--exclude` flag](/reference/node-selection/exclude). Use [custom commands](/docs/deploy/job-commands#advanced-ci-compare-changes) to modify the selection criteria.

Author: ```suggestion
      When you enable the **Run compare changes** checkbox, you can customize the comparison command to optimize your CI job. For example, if you have large models that take a long time to compare, you can exclude them to speed up the process using the [`--exclude` flag](/reference/node-selection/exclude). Use [custom commands](/docs/deploy/job-commands#compare-changes-custom-commands) to modify the selection criteria.
```

    - **dbt compare**<Lifecycle status="enterprise" /> &mdash; Enable this option to compare the last applied state of the production environment (if one exists) with the latest changes from the pull request, and identify what those differences are. To enable record-level comparison and primary key analysis, you must add a [primary key constraint](/reference/resource-properties/constraints) or [uniqueness test](/reference/resource-properties/data-tests#unique). Otherwise, you'll receive a "Primary key missing" error message in dbt Cloud.
    
      To review the comparison report, navigate to the [Compare tab](/docs/deploy/run-visibility#compare-tab) in the job run's details. A summary of the report is also available from the pull request in your Git provider (see the [CI report example](#example-ci-report)). 

      :::info Optimization tip 

>       When you enable the **Run compare changes** checkbox, you can customize the comparison command to optimize your CI job. For example, if you have large models that take a long time to compare, you can exclude them to speed up the process using the [`--exclude` flag](/reference/node-selection/exclude). Use [custom commands](/docs/deploy/job-commands#compare-changes-custom-commands) to modify the selection criteria.

Author: ```suggestion
      When you enable the **dbt compare** checkbox, you can customize the comparison command to optimize your CI job. For example, if you have large models that take a long time to compare, you can exclude them to speed up the process using the [`--exclude` flag](/reference/node-selection/exclude). Use [custom commands](/docs/deploy/job-commands#compare-changes-custom-commands) to modify the selection criteria.
```

>       :::

Author: ideally, i'd like to link to the event_time page that was created in this [pr here](https://github.com/dbt-labs/docs.getdbt.com/pull/6384) but i can add to it after it's merged

Author: also i know it's models/sources/seeds, etc. but once i link to the new event_time page, i'll remove the 'models' mention.

    
      To review the comparison report, navigate to the [Compare tab](/docs/deploy/run-visibility#compare-tab) in the job run's details. A summary of the report is also available from the pull request in your Git provider (see the [CI report example](#example-ci-report)). 

      :::info Optimization tip 
      When you enable the **dbt compare** checkbox, you can customize the comparison command to optimize your CI job. For example, if you have large models that take a long time to compare, you can exclude them to speed up the process using the [`--exclude` flag](/reference/node-selection/exclude). Refer to [compare changes custom commands](/docs/deploy/job-commands#compare-changes-custom-commands) for more details.

>       Additionally, if you set [`event_time`](/docs/build/incremental-microbatch#relevant-configs) in your models/seeds/snapshots/sources, it allows you to compare matching date ranges between tables by filtering to overlapping date ranges. This is useful for faster CI workflow or custom sampling set ups.

Author: ```suggestion
      Additionally, if you set [`event_time`](/reference/resource-configs/event-time) in your models/seeds/snapshots/sources, it allows you to compare matching date ranges between tables by filtering to overlapping date ranges. This is useful for faster CI workflow or custom sampling set ups.
```
